src/reflect/scala/reflect/internal/util/ScalaClassLoader.scala

```diff
@@ -150,4 +152,18 @@ object ScalaClassLoader {
   /** Finding what jar a clazz or instance came from */
   def originOfClass(x: Class[_]): Option[URL] =
     Option(x.getProtectionDomain.getCodeSource) flatMap (x => Option(x.getLocation))
+
+  private[this] val bootClassLoader: ClassLoader = {
```

Comment:
Since we only do this once, and Java versions are going to get weird next year, I wonder about simply always trying to call the method.

Reply:
"18.1", etc should still be >= "9", so I think this is safe.

We'll probably need to use this pattern more and more to support Java 9. We'll have to see how that plays out.

Sort of related thread:

https://twitter.com/giltene/status/908561954293215233

Example of just catching reflection errors after the fact rather than parsing `java.specication.version':

https://github.com/HdrHistogram/HdrHistogram/blob/master/src/main/java/org/HdrHistogram/Base64Helper.java

Reply:
Another example: https://github.com/real-logic/agrona/blob/master/agrona/src/main/java/org/agrona/hints/ThreadHints.java